wallet/src/bin/random-wallet-in-mem.rs

```diff
@@ -176,6 +240,47 @@ async fn create_wallet<'a>(
     )
 }
 
+fn add_balance(
+    addr: &UserAddress,
+    amount: u64,
+    asset: &AssetCode,
+    balances: &mut HashMap<UserAddress, HashMap<AssetCode, u64>>,
+) {
+    if !balances.contains_key(addr) {
+        balances.insert(addr.clone(), HashMap::new());
+    }
+
+    let assets = balances.get_mut(addr).unwrap();
+    let balance = *assets.get(asset).unwrap_or(&0);
+
+    assets.insert(*asset, balance + amount);
```

Comment:
This looks good. FYI though sometimes the `entry` API can make this kind of thing a bit clearer, like
```
let balance_ref: &mut u64 = balances
  .entry(addr)
  .or_default()
  .entry(asset)
  .or_default();
*balance_ref += amount;
```